library/alloc/src/string.rs

```diff
@@ -1553,18 +1553,21 @@ impl String {
         // Replace_range does not have the memory safety issues of a vector Splice.
         // of the vector version. The data is just plain bytes.
 
-        match range.start_bound() {
+        let start = range.start_bound();
+        match start {
             Included(&n) => assert!(self.is_char_boundary(n)),
             Excluded(&n) => assert!(self.is_char_boundary(n + 1)),
             Unbounded => {}
         };
-        match range.end_bound() {
+        let end = range.end_bound();
+        match end {
             Included(&n) => assert!(self.is_char_boundary(n + 1)),
             Excluded(&n) => assert!(self.is_char_boundary(n)),
             Unbounded => {}
         };
 
-        unsafe { self.as_mut_vec() }.splice(range, replace_with.bytes());
+        // Using `range` again would be unsound (#81138)
```

Comment:
Ah, here it is. Hmm.

Reply:
@bugadani Do you think another comment would be helpful? I added one where I thought it would make the most sense, since adding one above `let start` but not `let end` is a little weird.

Reply:
I think more comments wouldn't hurt 🙂

Reply:
That's true :stuck_out_tongue: